src/eager.js

```diff
@@ -6,8 +6,14 @@ import Helpers from './helpers';
 import Promise from './base/promise';
 import EagerBase from './base/eager';
 
-const getAttributeUnique = (models, attribute) =>
-  _.uniq(_.map(models, m => m.get(attribute)));
+const getAttributeUnique = (models, attribute) => {
+  return _.uniq(_.map(models, m => {
+    // use formatted attributes so that morphKey and foreignKey will match
+    // attribute keys
+    const attributes = m.format(_.clone(m.attributes));
```

Comment:
Yes my bad, it should be better to parse the specific attribute ? What do you think ?